proof/invariant-abstract/BCorres_AI.thy

```diff
@@ -16,6 +16,16 @@ abbreviation "bcorres \<equiv> bcorres_underlying truncate_state"
 
 abbreviation "s_bcorres \<equiv> s_bcorres_underlying truncate_state"
 
+context Arch begin arch_global_naming
+
+crunch arch_post_cap_deletion
+  for (bcorres) bcorres[wp]: truncate_state
+
+end
```

Comment:
Kinda neat that we can do this now in generic files, but also a little bit dangerous. Then again, it does remove duplication.

Reply:
It is a travesty, I agree. However, what we had before (`context begin interpretation Arch .`) was just as bad, and a lot slower :/

Reply:
This is definitely nicer.

Reply:
I don't think it's worth doing with the BCorres files already being a bit of a mess of dependencies, but would our current preference be to assume this in a locale and move the proof to the arch file?

Reply:
Yes, absolutely. I think this is an example of "no one likes BCorres, so it doesn't get the nice new toys". It got the worst approach. Then the next approach was the "Pre" thing, which also wasn't great, and then someone discovered the locale assumptions approach which I think is the one we should go with in any new proofs, barring maybe esoteric moments in `Invariants`-like files.